marklogic-data-hub-central/build.gradle

```diff
@@ -185,28 +184,17 @@ task copyUiFiles(type: Copy, dependsOn: deleteSpringBootUiFiles, group: taskGrou
     into springBootUiPath
 }
 
-// Before the Spring Boot jar is built, build the UI files and copy them over
-if (!(
-    gradle.startParameter.taskNames*.toLowerCase().contains("publishplugins") ||
-        gradle.startParameter.taskNames*.toLowerCase().contains("bintrayUpload") ||
-        gradle.startParameter.taskNames*.toLowerCase().contains("javadoc") ||
-        // bootRun doesn't need the UI, as it's assumed that the UI is being run separately via npm start
-        gradle.startParameter.taskNames*.toLowerCase().contains("bootrun") ||
-        gradle.startParameter.taskNames*.toLowerCase().contains("generateschemapojos") ||
-        project.hasProperty('skipui')
-)
-) {
+// When creating the HC war file, need to build the UI files and add them to the classpath so that the war file includes them
+if (gradle.startParameter.taskNames*.toLowerCase().contains("bootwar") || gradle.startParameter.taskNames*.toLowerCase().contains("buildrpm")) {
```

Comment:
Does bootRun need to be added here as well?

Reply:
@SameeraPriyathamTadikonda  What gradle task is being run for the cypress tests? And are the cypress tests expecting the middle tier to contain all of the UI code?

Reply:
I'll start hubcentral using the RPM and run tests using the npm command 'npm run cy:run'

Reply:
The cypress tests do expect the middle tier to contain all of the UI code. At least locally, I've had to build UI, run the server, and then run cypress tests using the cy:run command.